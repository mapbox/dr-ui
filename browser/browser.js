import React from 'react';
import PropTypes from 'prop-types';
import Icon from '@mapbox/mr-ui/icon';
class Browser extends React.PureComponent {
  render() {
    const {
      props
    } = this;
    return /*#__PURE__*/React.createElement("div", {
      className: "shadow-darken25 round bg-white"
    }, /*#__PURE__*/React.createElement("div", {
      className: "bg-gray h30-mm h18 round-t"
    }, /*#__PURE__*/React.createElement("div", {
      className: "color-white flex flex--center-cross h30-mm h18 px6"
    }, /*#__PURE__*/React.createElement("span", {
      className: "color-red-light"
    }, /*#__PURE__*/React.createElement(Icon, {
      name: "circle"
    })), /*#__PURE__*/React.createElement("span", {
      className: "color-yellow-light"
    }, /*#__PURE__*/React.createElement(Icon, {
      name: "circle"
    })), /*#__PURE__*/React.createElement("span", {
      className: "color-green-light"
    }, /*#__PURE__*/React.createElement(Icon, {
      name: "circle"
    })))), /*#__PURE__*/React.createElement("div", {
      className: "block-mm none"
    }, /*#__PURE__*/React.createElement("div", {
      className: "h30 bg-gray-lighter color-gray flex flex--space-between-main px6 flex--center-cross"
    }, /*#__PURE__*/React.createElement("div", {
      className: "inline-flex flex--space-between-main flex--center-cross mr18 w70"
    }, /*#__PURE__*/React.createElement(Icon, {
      name: "arrow-left"
    }), /*#__PURE__*/React.createElement(Icon, {
      name: "arrow-right"
    }), /*#__PURE__*/React.createElement(Icon, {
      name: "rotate"
    })), /*#__PURE__*/React.createElement("div", {
      className: "w120 h18 bg-white round-full flex-child-grow px12"
    }), /*#__PURE__*/React.createElement("div", {
      className: "inline-flex flex--space-between-main align-r flex--center-cross ml18 w70"
    }, /*#__PURE__*/React.createElement(Icon, {
      name: "heart"
    }), /*#__PURE__*/React.createElement(Icon, {
      name: "mapbox"
    }), /*#__PURE__*/React.createElement(Icon, {
      name: "menu"
    })))), props.children);
  }
}
export default Browser;