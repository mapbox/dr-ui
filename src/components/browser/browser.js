import React from 'react';
import PropTypes from 'prop-types';
import Icon from '@mapbox/mr-ui/icon';

class Browser extends React.PureComponent {
  render() {
    const { props } = this;
    return (
      <div className="shadow-darken25 round bg-white">
        <div className="bg-gray h30-mm h18 round-t">
          <div className="color-white flex-parent flex-parent--center-cross h30-mm h18 px6">
            <span className="color-red-light">
              <Icon name="circle" />
            </span>
            <span className="color-yellow-light">
              <Icon name="circle" />
            </span>
            <span className="color-green-light">
              <Icon name="circle" />
            </span>
          </div>
        </div>
        <div className="block-mm none">
          <div className="h30 bg-gray-faint color-gray flex-parent flex-parent--space-between-main px6 flex-parent--center-cross">
            <div
              style={{ width: '70px' }}
              className="flex-parent-inline flex-parent--space-between-main flex-parent--center-cross mr18"
            >
              <Icon name="arrow-left" />
              <Icon name="arrow-right" />
              <Icon name="rotate" />
            </div>
            <div className="w120 h18 bg-white round-full flex-child--grow px12" />
            <div
              style={{ width: '70px' }}
              className="flex-parent-inline flex-parent--space-between-main align-r flex-parent--center-cross ml18"
            >
              <Icon name="heart" />
              <Icon name="mapbox" />
              <Icon name="menu" />
            </div>
          </div>
        </div>
        {props.children}
      </div>
    );
  }
}

Browser.propTypes = {
  children: PropTypes.node.isRequired
};

export default Browser;
