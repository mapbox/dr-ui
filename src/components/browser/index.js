import main from './browser';

export default main;
