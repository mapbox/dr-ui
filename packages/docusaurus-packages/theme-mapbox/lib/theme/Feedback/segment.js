import forwardEvent from './forward-event';
import * as Sentry from '@sentry/browser';
import env from '@theme/AnalyticsShell/env';

// Send the event to Segment with forward event
export function sendToSegment({ state, props }) {
  const event = createSegmentEvent({ state, props });
  forwardEvent(event, props.webhook, (err) => {
    if (err) Sentry.captureException(err);
  });
}

// Creates event to send to Segment
function createSegmentEvent({ state, props }) {
  const { user, anonymousId, feedback, sessionId, category, categoryType } =
    state;
  const { site, section, location } = props;
  // set user if available else set anonymousId (needed for forward-event request)
  const identity = {
    ...(user && user.id ? { userId: user.id } : { anonymousId: anonymousId })
  };

  return {
    event: 'Sent docs feedback',
    ...identity,
    properties: {
      // the unique id for the feedbak session
      sessionId,
      // the feedback category that the user selected (helpful for yes, unhelpful for no)
      category,
      // the sub type for the category
      categoryType,
      // text feedback, if available
      ...(feedback && { feedback: feedback }),
      // name of current site (important for filtering in Mode)
      site,
      // (optional) name of section for longer pagers, helpful for fitering in Mode and identifying section areas
      section,
      // get page context
      page: location || undefined,
      // set user if available else set anonymousId (needed for Mode)
      userId: user && user.id ? user.id : undefined,
      anonymousId: anonymousId,
      // set plan, if available
      ...(user && user.plan && { plan: user.plan.id }),
      // get environment: staging or production
      environment: env()
    }
  };
}
