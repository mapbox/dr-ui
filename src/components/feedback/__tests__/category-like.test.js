import React from 'react';
import Feedback from '../feedback';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import {
  expectThankYou,
  textTooLong,
  textJustRight,
  textTooShort
} from './shared';
import forwardEvent from '../forward-event';
import * as Sentry from '@sentry/browser';

jest.mock('@sentry/browser');
jest.mock('../forward-event');

const SentryMockScope = {
  setTag: jest.fn(),
  setLevel: jest.fn(),
  setFingerprint: jest.fn()
};
Sentry.withScope.mockImplementation((callback) => {
  callback(SentryMockScope);
});

describe('Workflow', () => {
  afterAll(() => {
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });
  const feedback = mount(
    <Feedback
      site="dr-ui"
      location={{
        pathname: '/mapbox-gl-js/api/',
        hash: '#lnglat'
      }}
      webhook={{
        production: '',
        staging: ''
      }}
    />
  );

  describe('1 - Open feedback', () => {
    test('Click button, then set state', () => {
      const btn = feedback.find('button');
      btn.simulate('click');
      expect(feedback.state().isOpen).toBeTruthy();
    });
    test('Do not send data to Sentry', () => {
      expect(Sentry.init).not.toHaveBeenCalled();
    });
    test('Send data to Segment', () => {
      expect(forwardEvent).toHaveBeenCalledWith(
        {
          anonymousId: expect.anything(),
          event: 'Sent docs feedback',
          properties: {
            anonymousId: expect.anything(),
            category: undefined,
            categoryType: undefined,
            contactSupport: false,
            environment: 'staging',
            exited: false,
            helpful: undefined,
            page: { hash: '#lnglat', pathname: '/mapbox-gl-js/api/' },
            section: undefined,
            sessionId: expect.anything(),
            site: 'dr-ui'
          }
        },
        { production: '', staging: '' },
        expect.any(Function)
      );
    });
  });

  describe('2 - I like this page', () => {
    test('Select I like this page, then set state', () => {
      const btn = feedback.find('button[value="I like this page"]');
      btn.simulate('click', { target: { value: 'I like this page' } });
      expect(feedback.state().category).toEqual('I like this page');
      expect(feedback.state().helpful).toBeTruthy();
    });
    test('Check UI element state', () => {
      // The textarea is not available until the user selects an option
      expect(feedback.find('textarea').exists()).toBeFalsy();
      // The button is disabled until the user selects an option
      const submitButton = feedback.find('button#feedback-submit-button');
      expect(submitButton.props().disabled).toBeTruthy();
      expect(toJson(feedback)).toMatchSnapshot();
    });
    test('Do not send data to Sentry', () => {
      expect(Sentry.init).not.toHaveBeenCalled();
    });
    test('Send data to Segment', () => {
      expect(forwardEvent).toHaveBeenCalledWith(
        {
          anonymousId: expect.anything(),
          event: 'Sent docs feedback',
          properties: {
            anonymousId: expect.anything(),
            category: 'I like this page',
            categoryType: undefined,
            contactSupport: false,
            environment: 'staging',
            exited: false,
            helpful: true,
            page: { hash: '#lnglat', pathname: '/mapbox-gl-js/api/' },
            section: undefined,
            sessionId: expect.anything(),
            site: 'dr-ui'
          }
        },
        { production: '', staging: '' },
        expect.any(Function)
      );
    });
  });

  describe('3 - Select option', () => {
    test('Select option, then be able to enter text', () => {
      feedback
        .find('input[value="I found what I need"]')
        .props()
        .onChange({ target: { value: 'I found what I need' } });
      feedback.update();
      // The textarea is now available
      expect(feedback.find('textarea').exists()).toBeTruthy();
      // The button is enabled now
      const submitButton = feedback.find('button#feedback-submit-button');
      expect(submitButton.props().disabled).toBeFalsy();
    });
  });

  describe('4 - Enter text feedback', () => {
    test('Text over limit', () => textTooLong(feedback));
    test('Text under limit', () => textJustRight(feedback));
  });

  describe('5 - Submit feedback', () => {
    test('Click submit, set state', () => {
      const btn = feedback.find('#feedback-submit-button');
      btn.simulate('click');
      expect(feedback.state().sentFeedback).toBeTruthy();
      expect(feedback.state().category).toEqual('I like this page');
      expect(feedback.state().categoryType).toEqual('I found what I need');
      expect(feedback.state().feedback).toEqual(
        "I found a sandwich and I want to know why there isn't any mayonnaise."
      );
    });
    test('After submit, expect to see the thank you message', () => {
      expectThankYou(feedback);
    });

    test('Then send to Sentry', () => {
      expect(Sentry.init).toHaveBeenCalledWith({
        dsn: expect.anything(),
        environment: 'staging',
        maxValueLength: 1000
      });
      expect(SentryMockScope.setTag.mock.calls).toEqual([
        ['site', 'dr-ui'],
        ['category', 'I like this page'],
        ['categoryType', 'I found what I need'],
        ['referrer', ''],
        ['helpful', true]
      ]);
      expect(SentryMockScope.setLevel).toHaveBeenCalledWith('info');
      expect(SentryMockScope.setFingerprint).toHaveBeenCalledWith([
        'dr-ui',
        'I like this page',
        expect.anything(),
        expect.any(Date)
      ]);
      expect(Sentry.captureMessage).toHaveBeenCalledWith(
        "I found a sandwich and I want to know why there isn't any mayonnaise."
      );
    });

    test('Then send to Segment', () => {
      expect(forwardEvent).toHaveBeenCalledWith(
        {
          anonymousId: expect.anything(),
          event: 'Sent docs feedback',
          properties: {
            anonymousId: expect.anything(),
            category: 'I like this page',
            categoryType: 'I found what I need',
            contactSupport: false,
            environment: 'staging',
            exited: false,
            feedback:
              "I found a sandwich and I want to know why there isn't any mayonnaise.",
            helpful: true,
            page: { hash: '#lnglat', pathname: '/mapbox-gl-js/api/' },
            section: undefined,
            sessionId: expect.anything(),
            site: 'dr-ui'
          }
        },
        { production: '', staging: '' },
        expect.any(Function)
      );
    });
  });
});

describe('Workflow, Something else', () => {
  afterAll(() => {
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  const feedback = mount(
    <Feedback
      site="dr-ui"
      location={{
        pathname: '/mapbox-gl-js/api/',
        hash: '#lnglat'
      }}
      webhook={{
        production: '',
        staging: ''
      }}
    />
  );

  // Set up
  feedback.find('button').simulate('click');
  feedback
    .find('button[value="I like this page"]')
    .simulate('click', { target: { value: 'I like this page' } });

  describe('1 - Select option', () => {
    test('Select option', () => {
      feedback
        .find('input[value="Something else"]')
        .props()
        .onChange({ target: { value: 'Something else' } });
      feedback.update();
      // The textarea is now available
      expect(feedback.find('textarea').exists()).toBeTruthy();
      // The button is disabled (until text entered)
      const submitButton = feedback.find('button#feedback-submit-button');
      expect(submitButton.props().disabled).toBeTruthy();
    });
  });

  describe('2 - Enter text feedback', () => {
    test('Text over limit', () => textTooLong(feedback));
    test('Text under minimum', () => textTooShort(feedback));
    test('Text under limit', () => textJustRight(feedback));
  });

  describe('3 - Submit feedback', () => {
    test('Click submit, set state', () => {
      const btn = feedback.find('#feedback-submit-button');
      btn.simulate('click');
      expect(feedback.state().sentFeedback).toBeTruthy();
      expect(feedback.state().category).toEqual('I like this page');
      expect(feedback.state().categoryType).toEqual('Something else');
      expect(feedback.state().feedback).toEqual(
        "I found a sandwich and I want to know why there isn't any mayonnaise."
      );
    });
    test('After submit, expect to see the thank you message', () => {
      expectThankYou(feedback);
    });

    test('Then send to Sentry', () => {
      expect(Sentry.init).toHaveBeenCalledWith({
        dsn: expect.anything(),
        environment: 'staging',
        maxValueLength: 1000
      });
      expect(SentryMockScope.setTag.mock.calls).toEqual([
        ['site', 'dr-ui'],
        ['category', 'I like this page'],
        ['categoryType', 'Something else'],
        ['referrer', ''],
        ['helpful', true]
      ]);
      expect(SentryMockScope.setLevel).toHaveBeenCalledWith('info');
      expect(SentryMockScope.setFingerprint).toHaveBeenCalledWith([
        'dr-ui',
        'I like this page',
        expect.anything(),
        expect.any(Date)
      ]);
      expect(Sentry.captureMessage).toHaveBeenCalledWith(
        "I found a sandwich and I want to know why there isn't any mayonnaise."
      );
    });

    test('Then send to Segment', () => {
      expect(forwardEvent).toHaveBeenCalledWith(
        {
          anonymousId: expect.anything(),
          event: 'Sent docs feedback',
          properties: {
            anonymousId: expect.anything(),
            category: 'I like this page',
            categoryType: 'Something else',
            contactSupport: false,
            environment: 'staging',
            exited: false,
            feedback:
              "I found a sandwich and I want to know why there isn't any mayonnaise.",
            helpful: true,
            page: { hash: '#lnglat', pathname: '/mapbox-gl-js/api/' },
            section: undefined,
            sessionId: expect.anything(),
            site: 'dr-ui'
          }
        },
        { production: '', staging: '' },
        expect.any(Function)
      );
    });
  });
});

describe('Workflow, Select multiple; do not submit text feedback', () => {
  afterAll(() => {
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  const feedback = mount(
    <Feedback
      site="dr-ui"
      location={{
        pathname: '/mapbox-gl-js/api/',
        hash: '#lnglat'
      }}
      webhook={{
        production: '',
        staging: ''
      }}
    />
  );

  // Set up
  feedback.find('button').simulate('click');
  feedback
    .find('button[value="I like this page"]')
    .simulate('click', { target: { value: 'I like this page' } });

  describe('1 - Select option', () => {
    test('Select option, then be able to enter text', () => {
      feedback
        .find('input[value="Something else"]')
        .props()
        .onChange({
          target: { value: 'Something else,The information is accurate' }
        });
      feedback.update();
      // The textarea is now available
      expect(feedback.find('textarea').exists()).toBeTruthy();
      // The button is enabled now
      const submitButton = feedback.find('button#feedback-submit-button');
      expect(submitButton.props().disabled).toBeFalsy();
    });
  });

  describe('2 - Submit feedback', () => {
    test('Click submit, set state', () => {
      const btn = feedback.find('#feedback-submit-button');
      btn.simulate('click');
      expect(feedback.state().sentFeedback).toBeTruthy();
      expect(feedback.state().category).toEqual('I like this page');
      expect(feedback.state().categoryType).toEqual(
        'Something else,The information is accurate'
      );
      expect(feedback.state().feedback).toEqual('');
    });
    test('After submit, expect to see the thank you message', () => {
      expectThankYou(feedback);
    });

    test('Then send to Sentry', () => {
      expect(Sentry.init).toHaveBeenCalledWith({
        dsn: expect.anything(),
        environment: 'staging',
        maxValueLength: 1000
      });
      expect(SentryMockScope.setTag.mock.calls).toEqual([
        ['site', 'dr-ui'],
        ['category', 'I like this page'],
        ['categoryType', 'Something else,The information is accurate'],
        ['referrer', ''],
        ['helpful', true]
      ]);
      expect(SentryMockScope.setLevel).toHaveBeenCalledWith('info');
      expect(SentryMockScope.setFingerprint).toHaveBeenCalledWith([
        'dr-ui',
        'I like this page',
        expect.anything(),
        expect.any(Date)
      ]);
      expect(Sentry.captureMessage).toHaveBeenCalledWith('');
    });

    test('Then send to Segment', () => {
      expect(forwardEvent).toHaveBeenCalledWith(
        {
          anonymousId: expect.anything(),
          event: 'Sent docs feedback',
          properties: {
            anonymousId: expect.anything(),
            category: 'I like this page',
            categoryType: 'Something else,The information is accurate',
            contactSupport: false,
            environment: 'staging',
            exited: false,
            helpful: true,
            page: { hash: '#lnglat', pathname: '/mapbox-gl-js/api/' },
            section: undefined,
            sessionId: expect.anything(),
            site: 'dr-ui'
          }
        },
        { production: '', staging: '' },
        expect.any(Function)
      );
    });
  });
});
