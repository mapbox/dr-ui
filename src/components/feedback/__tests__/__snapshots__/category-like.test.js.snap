// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workflow 2 - I like this page Check UI element state 1`] = `
<Feedback
  feedbackSentryDsn="https://eccc8b561b9a461990309b01d33d54e3@sentry.io/1848287"
  location={
    Object {
      "hash": "#lnglat",
      "pathname": "/mapbox-gl-js/api/",
    }
  }
  site="dr-ui"
  type="page"
  webhook={
    Object {
      "production": "",
      "staging": "",
    }
  }
>
  <FeedbackWrapper
    helpful={true}
    title="I like this page"
  >
    <div
      className="dr-ui--feedback wmax300"
    >
      <div
        className="bg-gray-faint round py12 px12"
      >
        <div
          className="flex-parent flex-parent--column"
        >
          <div
            className="flex-child flex-parent flex-parent--space-between-main w-full mb12"
          >
            <div
              className="flex-child txt-bold"
            >
              I like this page
            </div>
            <div
              className="flex-child"
            >
              <Tooltip
                alignment="center"
                block={false}
                coloring="light"
                content="Close"
                disabled={false}
                maxWidth="medium"
                padding="small"
                placement="top"
                respondsToClick={false}
                textSize="s"
              >
                <PopoverTrigger
                  block={false}
                  content={
                    <div
                      className="txt-s wmax240"
                    >
                      Close
                    </div>
                  }
                  disabled={false}
                  popoverProps={
                    Object {
                      "alignment": "center",
                      "coloring": "light",
                      "hasPointer": true,
                      "hideWhenAnchorIsOffscreen": true,
                      "padding": "small",
                      "passthroughProps": Object {
                        "id": "tooltip-6",
                        "role": "tooltip",
                      },
                      "placement": "top",
                    }
                  }
                  receiveFocus={false}
                  renderHiddenContent={true}
                  respondsToClick={false}
                  respondsToFocus={true}
                  respondsToHover={true}
                  trapFocus={false}
                >
                  <div
                    className="inline-block"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                  >
                    <button
                      aria-describedby="tooltip-6"
                      aria-label="Close feedback"
                      className="link--gray"
                      id="feedback-close-button"
                      onClick={[Function]}
                    >
                      <Icon
                        inline={false}
                        name="close"
                        passthroughProps={Object {}}
                        size={18}
                      >
                        <svg
                          className="events-none icon"
                          focusable="false"
                          role="presentation"
                          style={
                            Object {
                              "height": 18,
                              "width": 18,
                            }
                          }
                        >
                          <use
                            xlinkHref="#icon-close"
                            xmlnsXlink="http://www.w3.org/1999/xlink"
                          />
                        </svg>
                      </Icon>
                    </button>
                    <div
                      className="hide-visually"
                      id="tooltip-6"
                      role="tooltip"
                    >
                      <div
                        className="txt-s wmax240"
                      >
                        Close
                      </div>
                    </div>
                  </div>
                </PopoverTrigger>
              </Tooltip>
            </div>
          </div>
          <div
            className="flex-child mb6 prose"
          >
            <CategoryLike
              leadText="Tell us what you like about this page."
              options={
                Array [
                  Object {
                    "label": "I found what I need",
                    "value": "I found what I need",
                  },
                  Object {
                    "label": "The information is accurate",
                    "value": "The information is accurate",
                  },
                  Object {
                    "label": "The page is easy to understand",
                    "value": "Easy to understand",
                  },
                  Object {
                    "label": "Something else",
                    "value": "Something else",
                  },
                ]
              }
              placeholder="What did you like?"
              submitFeedback={[Function]}
            >
              <p>
                Tell us what you like about this page.
              </p>
              <ControlCheckboxSet
                autoFocus={false}
                icon="check"
                id="feedback-category-like"
                legend="Select all that apply"
                onChange={[Function]}
                optional={false}
                options={
                  Array [
                    Object {
                      "label": "I found what I need",
                      "value": "I found what I need",
                    },
                    Object {
                      "label": "The information is accurate",
                      "value": "The information is accurate",
                    },
                    Object {
                      "label": "The page is easy to understand",
                      "value": "Easy to understand",
                    },
                    Object {
                      "label": "Something else",
                      "value": "Something else",
                    },
                  ]
                }
                themeCheckbox="mr6 inline-block checkbox--gray checkbox--s-label"
                themeCheckboxContainer="txt-s block mb6 flex-parent"
                value={Array []}
              >
                <ControlWrapper
                  id="feedback-category-like"
                  themeControlWrapper=""
                >
                  <div
                    className="relative "
                  >
                    <fieldset>
                      <ControlLegend
                        id="feedback-category-like"
                        optional={false}
                        text="Select all that apply"
                        themeLegend="txt-s txt-bold mb6"
                      >
                        <div>
                          <legend
                            className="inline-block txt-s txt-bold mb6"
                            htmlFor="feedback-category-like"
                          >
                            Select all that apply
                             
                          </legend>
                        </div>
                      </ControlLegend>
                      <div
                        data-test="feedback-category-like-input"
                        id="feedback-category-like"
                      >
                        <label
                          className="checkbox-container txt-s block mb6 flex-parent"
                          key="I found what I need"
                        >
                          <input
                            aria-checked={false}
                            autoFocus={false}
                            checked={false}
                            name="feedback-category-like"
                            onChange={[Function]}
                            role="checkbox"
                            type="checkbox"
                            value="I found what I need"
                          />
                          <div
                            className="mr6 inline-block checkbox--gray checkbox--s-label checkbox"
                          >
                            <Icon
                              inline={false}
                              name="check"
                              passthroughProps={Object {}}
                              size={18}
                            >
                              <svg
                                className="events-none icon"
                                focusable="false"
                                role="presentation"
                                style={
                                  Object {
                                    "height": 18,
                                    "width": 18,
                                  }
                                }
                              >
                                <use
                                  xlinkHref="#icon-check"
                                  xmlnsXlink="http://www.w3.org/1999/xlink"
                                />
                              </svg>
                            </Icon>
                          </div>
                          I found what I need
                        </label>
                        <label
                          className="checkbox-container txt-s block mb6 flex-parent"
                          key="The information is accurate"
                        >
                          <input
                            aria-checked={false}
                            autoFocus={false}
                            checked={false}
                            name="feedback-category-like"
                            onChange={[Function]}
                            role="checkbox"
                            type="checkbox"
                            value="The information is accurate"
                          />
                          <div
                            className="mr6 inline-block checkbox--gray checkbox--s-label checkbox"
                          >
                            <Icon
                              inline={false}
                              name="check"
                              passthroughProps={Object {}}
                              size={18}
                            >
                              <svg
                                className="events-none icon"
                                focusable="false"
                                role="presentation"
                                style={
                                  Object {
                                    "height": 18,
                                    "width": 18,
                                  }
                                }
                              >
                                <use
                                  xlinkHref="#icon-check"
                                  xmlnsXlink="http://www.w3.org/1999/xlink"
                                />
                              </svg>
                            </Icon>
                          </div>
                          The information is accurate
                        </label>
                        <label
                          className="checkbox-container txt-s block mb6 flex-parent"
                          key="Easy to understand"
                        >
                          <input
                            aria-checked={false}
                            autoFocus={false}
                            checked={false}
                            name="feedback-category-like"
                            onChange={[Function]}
                            role="checkbox"
                            type="checkbox"
                            value="Easy to understand"
                          />
                          <div
                            className="mr6 inline-block checkbox--gray checkbox--s-label checkbox"
                          >
                            <Icon
                              inline={false}
                              name="check"
                              passthroughProps={Object {}}
                              size={18}
                            >
                              <svg
                                className="events-none icon"
                                focusable="false"
                                role="presentation"
                                style={
                                  Object {
                                    "height": 18,
                                    "width": 18,
                                  }
                                }
                              >
                                <use
                                  xlinkHref="#icon-check"
                                  xmlnsXlink="http://www.w3.org/1999/xlink"
                                />
                              </svg>
                            </Icon>
                          </div>
                          The page is easy to understand
                        </label>
                        <label
                          className="checkbox-container txt-s block mb6 flex-parent"
                          key="Something else"
                        >
                          <input
                            aria-checked={false}
                            autoFocus={false}
                            checked={false}
                            name="feedback-category-like"
                            onChange={[Function]}
                            role="checkbox"
                            type="checkbox"
                            value="Something else"
                          />
                          <div
                            className="mr6 inline-block checkbox--gray checkbox--s-label checkbox"
                          >
                            <Icon
                              inline={false}
                              name="check"
                              passthroughProps={Object {}}
                              size={18}
                            >
                              <svg
                                className="events-none icon"
                                focusable="false"
                                role="presentation"
                                style={
                                  Object {
                                    "height": 18,
                                    "width": 18,
                                  }
                                }
                              >
                                <use
                                  xlinkHref="#icon-check"
                                  xmlnsXlink="http://www.w3.org/1999/xlink"
                                />
                              </svg>
                            </Icon>
                          </div>
                          Something else
                        </label>
                      </div>
                    </fieldset>
                    <div
                      role="alert"
                    />
                  </div>
                </ControlWrapper>
              </ControlCheckboxSet>
              <FeedbackButton
                disabled={true}
                onClick={[Function]}
              >
                <button
                  className="btn mt6"
                  disabled={true}
                  id="feedback-submit-button"
                  onClick={[Function]}
                >
                  Submit feedback
                </button>
              </FeedbackButton>
            </CategoryLike>
          </div>
        </div>
      </div>
    </div>
  </FeedbackWrapper>
</Feedback>
`;
